lib/src/repository.rs

```diff
@@ -60,22 +60,18 @@ pub trait RepositoryExt {
     ///
     fn issues(&self) -> Result<HeadRefsToIssuesIter>;
 
-    /// Create a new message
-    ///
-    /// This function creates a new issue message as well as an appropriate
-    /// reference. The oid of the new message will be returned.
-    /// The message will be part of the issue supplied by the caller. If no
-    /// issue is provided, a new issue will be initiated with the message.
-    /// In this case, the oid returned is also the oid of the new issue.
-    ///
-    fn create_message(&self,
-                      issue: Option<&Oid>,
-                      author: &Signature,
-                      committer: &Signature,
-                      message: &str,
-                      tree: &Tree,
-                      parents: &[&Commit]
-                     ) -> Result<Oid>;
+    /// Create a new issue with an initial message
+    ///
+    fn issue<'a, A, I, J>(&self,
```

Comment:
About the naming here: I'd suggest `create_issue` or something as function name, not only `issue` where one _has_ to look up the documentation on what it does. `create_issue` is a more speaking name, IMO.